src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -609,7 +609,7 @@ public void Dispose()
 #if !NATIVEAOT
         // This method is invoked by the VM when using the host-provided assembly load context
         // implementation.
-        private static Assembly? Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
+        private static RuntimeAssembly? Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
```

Comment:
Most of the changes are about changing the reverse FCall signatures to use RuntimeAssembly and asserting on unmanaged side that we really got `RuntimeAssembly`. I have done that to ensure that there are no other bugs like this.